pkg/registry/pod/etcd/etcd.go

```diff
@@ -280,7 +280,7 @@ func (r *ProxyREST) NewConnectOptions() (runtime.Object, bool, string) {
 }
 
 // Connect returns a handler for the pod proxy
-func (r *ProxyREST) Connect(ctx api.Context, id string, opts runtime.Object) (rest.ConnectHandler, error) {
+func (r *ProxyREST) Connect(ctx api.Context, id string, opts runtime.Object, _ rest.Responder) (http.Handler, error) {
```

Comment:
Updated, added a test to verify it invokes an error (didn't have one before)

On Mon, Oct 12, 2015 at 11:07 AM, Cesar Wong <notifications@github.com>
wrote:

> In pkg/registry/pod/etcd/etcd.go
> <https://github.com/kubernetes/kubernetes/pull/15442#discussion_r41766024>
> :
>
> > @@ -280,7 +280,7 @@ func (r *ProxyREST) NewConnectOptions() (runtime.Object, bool, string) {
> >  }
> >
> >  // Connect returns a handler for the pod proxy
> > -func (r *ProxyREST) Connect(ctx api.Context, id string, opts runtime.Object) (rest.ConnectHandler, error) {
> > +func (r *ProxyREST) Connect(ctx api.Context, id string, opts runtime.Object, _ rest.Responder) (http.Handler, error) {
>
> The responder object needs to be passed to the upgradeAwareProxyHandler so
> that it sets the error on it, instead of setting it on its own instance var:
>
>
> https://github.com/smarterclayton/kubernetes/blob/connect_responder/pkg/registry/generic/rest/proxy.go#L77
>
> https://github.com/smarterclayton/kubernetes/blob/connect_responder/pkg/registry/generic/rest/proxy.go#L110
>
> https://github.com/smarterclayton/kubernetes/blob/connect_responder/pkg/registry/generic/rest/proxy.go#L128-L150
>
> The RequestError method on that is also not needed anymore:
>
> https://github.com/smarterclayton/kubernetes/blob/connect_responder/pkg/registry/generic/rest/proxy.go#L63
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/15442/files#r41766024>.
>